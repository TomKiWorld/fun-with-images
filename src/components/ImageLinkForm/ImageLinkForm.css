.form {
  width: 100%;
  max-width: 43.75rem;
  min-width: 21.85rem;
  margin-bottom: 1rem;
}

.form .detect {
  background-color: #0239B7;
  border: 0;
  cursor: pointer;
  font-size: 1rem;
}

.form input {
  border: 0.1rem solid #0239B7;
}

@media screen and (min-width: 31rem) {
  .form {
    width: 80%;
  }
  
  .form .detect {
    font-size: 1.25rem;
  }
}
