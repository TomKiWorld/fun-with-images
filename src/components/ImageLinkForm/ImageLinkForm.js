import React from 'react';
import './ImageLinkForm.css';

/**
 * Image link form 
 * - Input field to submit a url
 * - submit button
 * 
 * Required props:
 * - onInputChange => Register input changes to state
 * - onButtonSubmit => Submit the image url
 * - imageUrlError => Display error to screen
 * - inputValue => Input value from state important in case of resubmission 
 */
// Allow pressing the enter key and not only the button
const handleKeyUp = (e) => {
  if (e.keyCode === 13) {
    const detect = document.querySelector('.detect');
    detect.click();
  }
}

const ImageLinkForm = ({ onInputChange, onButtonSubmit, imageUrlError, getColorsError, getFacesError, inputValue }) => {
  return (
    <div>
      <p className='f3'>This App detects faces in images</p>
      <p>Click on the <strong>paint icon</strong> to see which colors were detected</p>
      <div className='center'>
        <div className='center form pa4 br3 shadow-5'>
          <input 
            className='f4 pa2 w-70 center' 
            value={inputValue}
            type='url' 
            onChange={onInputChange}
            onKeyUp={(e) => handleKeyUp(e)}
          />
          <button 
            className='detect w-30 grow link ph3 pv2 dib white bg-light-blue' 
            onClick={onButtonSubmit}
          >Detect</button>
        </div>
      </div>
        {imageUrlError}
        {getColorsError}
        {getFacesError}
    </div>
  );
}

export default ImageLinkForm;
