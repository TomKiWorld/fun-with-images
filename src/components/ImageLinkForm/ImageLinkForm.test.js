import { shallow } from 'enzyme';
import React from 'react';
import ImageLinkForm from './ImageLinkForm';

describe('ImageLinkForm Component', () => {
  const mockProps = {
    value: '',
    clicked: false
  };
  const mockChange = (value) => {
    mockProps.value= value;
  };
  const wrapper = shallow(
    <ImageLinkForm 
      onInputChange={(value) => mockChange(value)}
    />
  );
  it('Expect to render ImageLinkForm Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Expect to react to change', () => {
    const urlInput =wrapper.find('input').at(0);
    urlInput.simulate('change', { target: { value: 'image-url'} });
    expect(mockProps.value).toBe('image-url');
  });
});
