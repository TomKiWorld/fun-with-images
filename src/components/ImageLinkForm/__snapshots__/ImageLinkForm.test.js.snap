// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageLinkForm Component Expect to render ImageLinkForm Component 1`] = `ShallowWrapper {}`;
